crates/storage-rocksdb/src/lib.rs

```diff
@@ -318,96 +244,55 @@ impl RocksDBStorage {
-            ),
-            rocksdb::ColumnFamilyDescriptor::new(
-                cf_name(State),
-                cf_options(rock_opts, cache.clone()),
-            ),
-            rocksdb::ColumnFamilyDescriptor::new(
-                cf_name(InvocationStatus),
-                cf_options(rock_opts, cache.clone()),
-            ),
-            rocksdb::ColumnFamilyDescriptor::new(
-                cf_name(ServiceStatus),
-                cf_options(rock_opts, cache.clone()),
-            ),
-            rocksdb::ColumnFamilyDescriptor::new(
-                cf_name(Journal),
-                cf_options(rock_opts, cache.clone()),
-            ),
-            rocksdb::ColumnFamilyDescriptor::new(
-                cf_name(Idempotency),
-                cf_options(rock_opts, cache.clone()),
-            ),
+            (CfName::new(cf_name(Inbox)), cf_options()),
+            (CfName::new(cf_name(State)), cf_options()),
+            (CfName::new(cf_name(InvocationStatus)), cf_options()),
+            (CfName::new(cf_name(ServiceStatus)), cf_options()),
+            (CfName::new(cf_name(Journal)), cf_options()),
+            (CfName::new(cf_name(Idempotency)), cf_options()),
             //
             // keyed by partition id + suffix
             //
-            rocksdb::ColumnFamilyDescriptor::new(
-                cf_name(Outbox),
-                cf_options(rock_opts, cache.clone()),
-            ),
-            rocksdb::ColumnFamilyDescriptor::new(
-                cf_name(Timers),
-                cf_options(rock_opts, cache.clone()),
-            ),
+            (CfName::new(cf_name(Outbox)), cf_options()),
+            (CfName::new(cf_name(Timers)), cf_options()),
             // keyed by partition_id + partition_id
-            rocksdb::ColumnFamilyDescriptor::new(
-                cf_name(Deduplication),
-                cf_options(rock_opts, cache.clone()),
-            ),
+            (CfName::new(cf_name(Deduplication)), cf_options()),
             // keyed by partition_id + u64
-            rocksdb::ColumnFamilyDescriptor::new(
-                cf_name(PartitionStateMachine),
-                cf_options(rock_opts, cache.clone()),
-            ),
+            (CfName::new(cf_name(PartitionStateMachine)), cf_options()),
         ];
 
-        let db_options = db_options(rock_opts);
-
-        let is_empty_db = Self::is_empty_db(options.data_dir());
-
-        let rdb = DB::open_cf_descriptors(&db_options, options.data_dir(), tables)
-            .map_err(BuildError::from_rocksdb_error)?;
+        let db_spec = DbSpec::new_optimistic_db(
+            DbName::new(DB_NAME),
+            Owner::PartitionProcessor,
+            data_dir,
+            db_options(),
+            cfs,
+        );
+        // todo remove this when open_db is async
+        let rdb = tokio::task::spawn_blocking(move || {
+            RocksDbManager::get().open_db(updateable_opts, db_spec)
+        })
+        .await
```

Comment:
Is the idea to make `open_db` async so that we can dynamically open new dbs w/o blocking a tokio thread when we start new PPs in the future?

Reply:
Yes, hopefully.